### PR TITLE
Use lld and compiler-rt

### PR DESCRIPTION
**Pull request by arunthomas**

### CODE REVIEW

File: freedom-e-sdk/bsp/env/start.S [hunk 1]
```diff
@@ -70,10 +70,12 @@ _start:
 	/* Call global constructors */
 	//la a0, __libc_fini_array
 1:
+/*
```
Comment (natestuder) [inline freedom-e-sdk/bsp/env/start.S:73]: So this is removing the libc_init_array, which I presume llvm doesn't have, but what do the other changes have to do with LLVM?

File: freedom-e-sdk/bsp/env/freedom-e300-hifive1/flash.lds [hunk 2]
```diff
@@ -155,7 +155,7 @@ SECTIONS
   .stack ORIGIN(ram) + LENGTH(ram) - __stack_size :
   {
     PROVIDE( _heap_end = . );
-    . = __stack_size;
+    . += __stack_size;
```
Comment (natestuder) [inline freedom-e-sdk/bsp/env/freedom-e300-hifive1/flash.lds:158]: That might be why the stack_size setting never seemed to work.

Comment (arunthomas) [reply freedom-e-sdk/bsp/env/freedom-e300-hifive1/flash.lds:158]: Yeah, I think so too.

File: freedom-e-sdk/bsp/env/start.S (hunk 1 above)
Comment (arunthomas) [reply freedom-e-sdk/bsp/env/start.S:73]: lld does not have the I (initialized) attribute, so I had to remove it from the linker script.

File: freedom-e-sdk/bsp/env/freedom-e300-hifive1/flash.lds (hunk 2 above)
Comment (arunthomas) [reply freedom-e-sdk/bsp/env/freedom-e300-hifive1/flash.lds:158]: To clarify, this fixes a bug in the original linker script. GNU ld accepts this, but lld rightly throws an error.

**Pull request closed without merging**